internal/monitors/device-updater/device_updater.go

```diff
@@ -47,42 +50,119 @@ func (d *DeviceUpdater) UpdateDevices() {
 		return
 	}
 
-	for _, fleet := range fleets {
+	for fleetIndex := range fleets {
```

Comment:
I think its easier to reason about the fleet directly why use the index directly?